### PR TITLE
ARROW-12972: [CI] Fix centos-8 cmake error

### PR DESCRIPTION
**Pull request by cyb70289**
Install libarchive to fix cmake error on centos-8.

cmake: undefined symbol: archive_write_add_filter_zstd

### CODE REVIEW

**Comment by cyb70289** [comment]
@kou , anyway to trigger related crossbow build?

**Comment by kou** [comment]
@github-actions crossbow submit centos-8-*

**Comment by cyb70289** [comment]
@github-actions crossbow submit centos-8-*

File: dev/tasks/linux-packages/apache-arrow/yum/centos-8/Dockerfile [hunk 1]
```diff
@@ -40,6 +40,7 @@ RUN \
     glog-devel \
     gobject-introspection-devel \
     gtk-doc \
+    libarchive \
```
Comment (kou) [inline dev/tasks/linux-packages/apache-arrow/yum/centos-8/Dockerfile:43]: We don't need this, right?

**Review by kou — APPROVED** [review]
+1

Thanks!

**Comment by kou** [comment]
The centos-8-arm64 failure is another problem.

File: dev/tasks/linux-packages/apache-arrow/yum/centos-8/Dockerfile (hunk 1 above)
Comment (cyb70289) [reply dev/tasks/linux-packages/apache-arrow/yum/centos-8/Dockerfile:43]: Yes, will remove.

Comment (kou) [reply dev/tasks/linux-packages/apache-arrow/yum/centos-8/Dockerfile:43]: We don't need this, could you revert this before we merge this?

Comment (kou) [reply dev/tasks/linux-packages/apache-arrow/yum/centos-8/Dockerfile:43]: Thanks!

Comment (cyb70289) [reply dev/tasks/linux-packages/apache-arrow/yum/centos-8/Dockerfile:43]: Some explanation:

`libarchive` is pre-installed in `arm64v8/centos:8`, but not in `centos:8`.
This dockerfile is based on `arm64v8/centos:8`, so it's okay.
`verify-yum.sh` is based on `centos:8` (multi arch image) and fails.

Will we unify x86 and arm to use `centos:8` only?

**Comment by kou** [comment]
@github-actions crossbow submit centos-8-*

**Comment by cyb70289** [comment]
I don't understand why x86 doesn't have this issue. Per my test with `centos:8` image, both x86 and Arm have this cmake undefined symbol error.

**Comment by kou** [comment]
> I don't understand why x86 doesn't have this issue.

This issue is "cmake: undefined symbol: archive_write_add_filter_zstd" not pygit2 error https://travis-ci.com/github/ursacomputing/crossbow/builds/228149814#L5782 , right?

This issue happens on x86 without this fix: https://github.com/ursacomputing/crossbow/runs/2760533808?check_suite_focus=true#step:11:775

**Comment by cyb70289** [comment]
So `libarchive` is necessary in both places. Will add back.

**Comment by kou** [comment]
Oh, sorry.

**Comment by cyb70289** [comment]
No worry. It's trivial. :)

**Comment by cyb70289** [comment]
@github-actions crossbow submit centos-8-*

**Pull request closed without merging**